src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/components/dot-content-type-fields-variables-table-row/dot-content-type-fields-variables-table-row.component.html

```diff
@@ -1,32 +1,46 @@
-<tr class="field-variable-row" [ngClass]="{'field-variable-row-highlight': rowActiveHighlight}">
-    <td #keyCell [pEditableColumn]="fieldVariable.key" pEditableColumnField="key">
-        <p-cellEditor>
-            <ng-template pTemplate="input">
-                <input pInputText
-                    class="field-variable-key-input"
-                    type="text"
-                    [placeholder]="messages['contenttypes.field.variables.key_input.placeholder']"
-                    (blur)="rowActiveHighlight = false"
-                    (focus)="editFieldInit()"
-                    (keyup)="editFieldInit()"
-                    (keydown.escape)="onCancel($event)"
-                    (keydown.enter)="onPressEnter($event)"
-                    [(ngModel)]="fieldVariable.key">
-            </ng-template>
-            <ng-template pTemplate="output">
-                <span *ngIf="fieldVariable.key">
-                    {{ fieldVariable.key }}
-                </span>
-                <span class="empty-placeholder" *ngIf="!fieldVariable.key">
-                    {{ messages['contenttypes.field.variables.key_input.placeholder'] }}
-                </span>
-            </ng-template>
-        </p-cellEditor>
-    </td>
+<tr class="field-variable-row" [ngClass]="{ 'field-variable-row-highlight': rowActiveHighlight }">
+    <ng-template [ngIf]="fieldVariable.key" [ngIfElse]="editable">
+        <td>
+            {{ fieldVariable.key }}
+        </td>
+    </ng-template>
+    <ng-template #editable>
+        <td #keyCell [pEditableColumn]="fieldVariable.key" pEditableColumnField="key">
+            <p-cellEditor>
+                <ng-template pTemplate="input">
+                    <input
+                        pInputText
+                        class="field-variable-key-input"
+                        type="text"
+                        [disabled]="keyDisabled"
+                        [placeholder]="
+                            messages['contenttypes.field.variables.key_input.placeholder']
+                        "
+                        (blur)="rowActiveHighlight = false"
+                        (focus)="editFieldInit()"
+                        (keyup)="editFieldInit()"
+                        (keydown.escape)="onCancel($event)"
+                        (keydown.enter)="onPressEnter($event)"
+                        [(ngModel)]="fieldVariable.key"
+                    />
+                </ng-template>
+                <ng-template pTemplate="output">
+                    <span *ngIf="fieldVariable.key">
+                        {{ fieldVariable.key }}
+                    </span>
+                    <span class="empty-placeholder" *ngIf="!fieldVariable.key">
+                        {{ messages['contenttypes.field.variables.key_input.placeholder'] }}
+                    </span>
+                </ng-template>
+            </p-cellEditor>
```

Comment:
maybe this code should be a componente, what do you think?

Reply:
That’s a component